app/src/main/java/pl/ludek/smat/home/pool_office_client/presentation/MainActivity.kt

```diff
@@ -0,0 +1,134 @@
+import pl.ludek.smat.home.pool_office_client.domain.model.PoolInfoData
+
+class MainActivity : AppCompatActivity(), View.OnClickListener {
+    private lateinit var relayFirstOnStr:String    // "RELAY-FIRST-ON"
+    private lateinit var  relayFirstOffStr:String  // "RELAY-FIRST-OFF"
+    private lateinit var relayFiveOnStr:String    // "RELAY-FIVE-ON"
+    private lateinit var relayFiveOffStr:String  // "RELAY-FIVE-OFF"
+    private lateinit var relayAllOnStr:String    // "RELAY-ALL-ON"
+    private lateinit var relayAllOffStr:String   // "RELAY-ALL-OFF"
+    private lateinit var errorRelayStr:String
+    private lateinit var  errorSensorStr:String
+    private lateinit var dataFromRepository: PoolInfoViewModel
+    private lateinit var dataFromRelay: RelayViewModel
+    private val dataFromSensor: SensorsViewModel by viewModels()
+    private lateinit var sensorDataTextView: TextView
+    private lateinit var relayFirstOnButton:Button
+    private lateinit var relayFiveOnButton: Button
+    private lateinit var relayAllOnButton: Button
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        relayFirstOnStr = getString(R.string.relay_1_on)
+        relayFirstOffStr = getString(R.string.relay_1_off)
+        relayFiveOnStr = getString(R.string.relay_5_on)
+        relayFiveOffStr = getString(R.string.relay_5_off)
+        relayAllOnStr = getString(R.string.relay_all_on)
+        relayAllOffStr = getString(R.string.relay_all_off)
+        errorRelayStr = getString(R.string.relay_error)
+        errorSensorStr = getString(R.string.data_from_sensor)
+        relayFirstOnButton = findViewById(R.id.relayOneOn)
+        relayFiveOnButton = findViewById(R.id.relayFieveOn)
+        relayAllOnButton = findViewById(R.id.relayAllOn)
+        dataFromRepository = ViewModelProvider(this).get(PoolInfoViewModel::class.java)
+        dataFromRelay = ViewModelProvider(this).get(RelayViewModel::class.java)
+        sensorDataTextView = findViewById(R.id.sensorDataView)
+
+        val dataFromSensorObserver = Observer<PoolInfoData>{ inputDataFromSensor ->
+            if(inputDataFromSensor.p1 == 0.0f && inputDataFromSensor.t1 == 0.0f && inputDataFromSensor.t2 == 0.0f && inputDataFromSensor.t3 == 0.0f){
+                    sensorDataTextView.text = errorSensorStr
+            }else {
+                val rez:String =" T1= " + inputDataFromSensor.t1 +
+                        " T2= " + inputDataFromSensor.t2 +
+                        " T3= " + inputDataFromSensor.t3 +
+                        " P1= " + inputDataFromSensor.p1
+                sensorDataTextView.text = rez
+            }
+        }
+        dataFromSensor.currentSensorData.observe(this,dataFromSensorObserver)
+        setDataToLiveData()
+        relayFirstOnButton.setOnClickListener(this)
+        relayFiveOnButton.setOnClickListener(this)
+        relayAllOnButton.setOnClickListener(this)
```

Comment:
Avoid using global listeners.
It is much better to have the separate anonymous classes. With Kotlin you should lambda expressions like this:
```
relayFirstOnButton.setOnClickListener {
    if (relayFirstOnButton.text == relayFirstOnStr) {
        dataFromRelay.postOneRelayOn().observe(this, Observer {
            if (!it.errorRelay) {
                relayFirstOnButton.text = relayFirstOffStr
            } else {
                showToast(errorRelayStr)
            }
        })
    } else {
        dataFromRelay.postOneRelayOff().observe(this, Observer {
            if (!it.errorRelay) {
                relayFirstOnButton.text = relayFirstOnStr
            } else {
                showToast(errorRelayStr)
            }
        })
    }
}
```
Also, in order to make the code more readable it is good to move the views initiation from `onCreate()` to dedicated method (for instance `fun initView()`) called from `onCreate()`.